ast_utils/color.c2

```diff
@@ -16,25 +16,73 @@
 module color;
 import unistd;
 
-public const char[] Black    = "\033[0;30m";
-public const char[] Red      = "\033[0;31m";
-public const char[] Green    = "\033[0;32m";
-public const char[] Yellow   = "\033[0;33m";
-public const char[] Blue     = "\033[0;34m";
-public const char[] Magenta  = "\033[0;35m";
-public const char[] Cyan     = "\033[0;36m";
-public const char[] Grey     = "\033[0;37m";
-public const char[] Darkgrey = "\033[01;30m";
-public const char[] Bred     = "\033[01;31m";
-public const char[] Bgreen   = "\033[01;32m";
-public const char[] Byellow  = "\033[01;33m";
-public const char[] Bblue    = "\033[01;34m";
-public const char[] Bmagenta = "\033[01;35m";
-public const char[] Bcyan    = "\033[01;36m";
-public const char[] White    = "\033[01;37m";
-public const char[] Normal   = "\033[0m";
+public type Color enum u8 {
+    None,
+    Black,
+    Red,
+    Green,
+    Yellow,
+    Blue,
+    Magenta,
+    Cyan,
+    Grey,
+    Darkgrey,
+    Bred,
+    Bgreen,
+    Byellow,
+    Bblue,
+    Bmagenta,
+    Bcyan,
+    White,
+    Normal,
+}
+
+public const Color Black    = Color.Black;
+public const Color Red      = Color.Red;
+public const Color Green    = Color.Green;
+public const Color Yellow   = Color.Yellow;
+public const Color Blue     = Color.Blue;
+public const Color Magenta  = Color.Magenta;
+public const Color Cyan     = Color.Cyan;
+public const Color Grey     = Color.Grey;
+public const Color Darkgrey = Color.Darkgrey;
+public const Color Bred     = Color.Bred;
+public const Color Bgreen   = Color.Bgreen;
+public const Color Byellow  = Color.Byellow;
+public const Color Bblue    = Color.Bblue;
+public const Color Bmagenta = Color.Bmagenta;
+public const Color Bcyan    = Color.Bcyan;
+public const Color White    = Color.White;
+public const Color Normal   = Color.Normal;
+
+const char*[elemsof(Color)] defaultColors = {
+    [Color.None]     = "",
+    [Color.Black]    = "\033[0;30m",
+    [Color.Red]      = "\033[0;31m",
+    [Color.Green]    = "\033[0;32m",
+    [Color.Yellow]   = "\033[0;33m",
+    [Color.Blue]     = "\033[0;34m",
+    [Color.Magenta]  = "\033[0;35m",
+    [Color.Cyan]     = "\033[0;36m",
+    [Color.Grey]     = "\033[0;37m",
+    [Color.Darkgrey] = "\033[1;30m",
+    [Color.Bred]     = "\033[1;31m",
+    [Color.Bgreen]   = "\033[1;32m",
+    [Color.Byellow]  = "\033[1;33m",
+    [Color.Bblue]    = "\033[1;34m",
+    [Color.Bmagenta] = "\033[1;35m",
+    [Color.Bcyan]    = "\033[1;36m",
+    [Color.White]    = "\033[1;37m",
+    [Color.Normal]   = "\033[0m",
+}
+
+bool hasColor;
 
 public fn bool useColor() {
-    return unistd.isatty(1);
+    return hasColor = unistd.isatty(1);
```

Comment:
ehhh this line does not parse well for me.. I think making it 2 lines is more readable.
hasColor = ..;
return hasColor;

Reply:
I removed the `hasColor` global variable.